code/controllers/CMSPageHistoryController.php

```diff
@@ -169,17 +234,19 @@ public function getEditForm($id = null, $fields = null, $versionID = null, $comp
 			"ID" => $id,
 			"Version" => $versionID,
 		));
-		
-		if(($record && $record->isLatestVersion())) {
+
+		if ($record->isLatestVersion()) {
 			$revert->setReadonly(true);
 		}
-		
+
 		$form->removeExtraClass('cms-content');
 
+		$form->setFormAction(Controller::join_links($form->FormAction(), $id, $versionID));
```

Comment:
The risk here is that $form->FormAction() Could already have the ID in it set by CMSmain::getEditForm(). Repeating this action could end up with double-ID.

Safer fix here is.

```php
$form->setFormAction(Controller::join_links($this->Link(), $form->getName(), $id, $versionID));
```